lib/visualization.js

```diff
@@ -520,6 +556,136 @@ class MDPVis extends Visualization {
 			.attr('cy', y);
 	}
 }
+ //Time inconsistency vis
+class TIVis extends Visualization {
+	constructor(env, trace) {
+		super(trace);
+		this.time = 0;
+		this.current_state_no = env.config.initial_state;
+		this.states = env.states;
+		this.plan_vis = [];
+		this.history = trace.states;
+		this.counter_goal = env.delayed_dispense;
+		this.counters = [];
+		this.plan_trace = trace.plan_trace
+		for (let i = 0; i < this.history.length; i++){
+			this.counters.push(this.history[i].count)
+		}
+		this.pos_array = env.pos_array
+		let gridvis = document.getElementById('gridvis');
+		//TODO fix magic numbers (maybe base canvas size on state locations)
+		gridvis.style.display = 'block';
+
+		//Get number of time inconsistent actions and output their occurences
```

Comment:
keep it here for now, but we're going to need to move this to another function, added to the callback at the end of the demo.